pkg/debug/start_debug.go

```diff
@@ -0,0 +1,167 @@
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	debug_deployment := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-debug", deploymentName),
+			Namespace: clusterNamespace,
+			Labels:    labels,
+		},
+		Spec: deploymentSpec,
+	}
+
+	debugDeployment, err := clientset.AppsV1().Deployments(clusterNamespace).Create(context.TODO(), debug_deployment, v1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating deployment %s. %v\n", debug_deployment, err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("ensure the debug deployment %s is scaled up\n", deploymentName)
+
+	if err := getAndUpdatetheScale(clientset, clusterNamespace, debugDeployment.Name, 1); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+}
+
+func getAndUpdatetheScale(clientset *k8s.Clientset, clusterNamespace, deploymentName string, scaleCount int) error {
+	scale, err := clientset.AppsV1().Deployments(clusterNamespace).GetScale(context.TODO(), deploymentName, v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get scale of deployment %s. %v\n", deploymentName, err)
+	}
+
+	scale.Spec.Replicas = int32(scaleCount)
+	_, err = clientset.AppsV1().Deployments(clusterNamespace).UpdateScale(context.TODO(), deploymentName, scale, v1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update scale of deployment %s. %v\n", deploymentName, err)
+	}
+	return nil
+}
+
+func verifyDeploymentExists(clientset *k8s.Clientset, clusterNamespace, deploymentName string) (*appsv1.Deployment, error) {
+	deployment, err := clientset.AppsV1().Deployments(clusterNamespace).Get(context.TODO(), deploymentName, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return deployment, nil
+}
+
+func updateDeployment(clientset *k8s.Clientset, clusterNamespace string, deploymentName *appsv1.Deployment) error {
+	_, err := clientset.AppsV1().Deployments(clusterNamespace).Update(context.TODO(), deploymentName, v1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func podRunningWithLabel(clientset *k8s.Clientset, clusterNamespace string, deploymentSpec appsv1.DeploymentSpec) (string, error) {
+
+	pods, err := clientset.CoreV1().Pods(clusterNamespace).List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Labels["ceph_daemon_type"] == deploymentSpec.Template.Labels["ceph_daemon_type"] &&
+			pod.Labels["ceph_daemon_id"] == deploymentSpec.Template.Labels["ceph_daemon_id"] {
+			return pod.Name, nil
+		}
+	}
+	return "", fmt.Errorf("No pod with labels matching %s:%s", deploymentSpec.Template.Labels, deploymentSpec.Template.Labels)
+}
+
+func waitForPodDeletion(clientset *k8s.Clientset, clusterNamespace, podName string) error {
+	for i := 0; i < 60; i++ {
+		err := clientset.CoreV1().Pods(clusterNamespace).Delete(context.TODO(), podName, v1.DeleteOptions{})
+		if err != nil {
+			return nil
+		}
+		time.Sleep(time.Second * 5)
```

Comment:
Whenever we are waiting, let's print a message so they know something is happening

Reply:
done